samza-kv/src/test/scala/org/apache/samza/storage/kv/TestLoggedStore.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.storage.kv
+
+import java.util._
+
+import org.apache.samza.config.Config
+import org.apache.samza.metrics.Counter
+import org.apache.samza.system.SystemStreamPartition
+import org.apache.samza.task.MessageCollector
+import org.apache.samza.{Partition, SamzaException}
+import org.junit.{Before, Test}
+import org.mockito.Matchers
+import org.mockito.Mockito._
+
+
+class TestLoggedStore {
+
+  val store = mock(classOf[KeyValueStore[Array[Byte], Array[Byte]]])
```

Comment:
Has this been tested with a real store implementation? Most samza jobs do not create KVStore<byte[], byte[]>, they create something typed and then after SerializedKVStore serializes the entries, it becomes byte[], byte[]. 

It would be good to ensure the types are compatible in a real scenario with each of the KVStore wrappers.

Reply:
I've changed the implementation to have a dedicated LargeMessageStore that runs on 2 configs given by the user to check for large messages and handle them, and added the necessary integration tests for this. I've added more information about this in the description.